typescript/utils/src/altvm.ts

```diff
@@ -0,0 +1,663 @@
+  signer: string;
+  ismId: string;
+  newOwner: string;
+};
+export type ResSetRoutingIsmOwner = {
+  newOwner: string;
+};
+
+export type ReqCreateNoopIsm = {
+  signer: string;
+};
+export type ResCreateNoopIsm = {
+  ismId: string;
+};
+
+export type ReqCreateMerkleTreeHook = { signer: string; mailboxId: string };
+export type ResCreateMerkleTreeHook = {
+  hookId: string;
+};
+
+export type ReqCreateInterchainGasPaymasterHook = {
+  signer: string;
+  denom: string;
+};
+export type ResCreateInterchainGasPaymasterHook = {
+  hookId: string;
+};
+
+export type ReqSetInterchainGasPaymasterHookOwner = {
+  signer: string;
+  hookId: string;
+  newOwner: string;
+};
+export type ResSetInterchainGasPaymasterHookOwner = {
+  newOwner: string;
+};
+
+export type ReqSetDestinationGasConfig = {
+  signer: string;
+  hookId: string;
+  destinationGasConfig: {
+    remoteDomainId: number;
+    gasOracle: {
+      tokenExchangeRate: string;
+      gasPrice: string;
+    };
+    gasOverhead: string;
+  };
+};
+export type ResSetDestinationGasConfig = {
+  destinationGasConfig: {
+    remoteDomainId: number;
+    gasOracle: {
+      tokenExchangeRate: string;
+      gasPrice: string;
+    };
+    gasOverhead: string;
+  };
+};
+
+export type ReqCreateValidatorAnnounce = {
+  signer: string;
+  mailboxId: string;
+};
+export type ResCreateValidatorAnnounce = {
+  validatorAnnounceId: string;
+};
+
+// ### POPULATE WARP ###
+
+export type ReqCreateCollateralToken = {
+  signer: string;
+  mailboxId: string;
+  originDenom: string;
+};
+export type ResCreateCollateralToken = {
+  tokenId: string;
+};
+
+export type ReqCreateSyntheticToken = {
+  signer: string;
+  mailboxId: string;
+};
+export type ResCreateSyntheticToken = {
+  tokenId: string;
+};
+
+export type ReqSetTokenOwner = {
+  signer: string;
+  tokenId: string;
+  newOwner: string;
+};
+export type ResSetTokenOwner = {
+  newOwner: string;
+};
+
+export type ReqSetTokenIsm = {
+  signer: string;
+  tokenId: string;
+  ismId: string;
+};
+export type ResSetTokenIsm = {
+  ismId: string;
+};
+
+export type ReqEnrollRemoteRouter = {
+  signer: string;
+  tokenId: string;
+  remoteRouter: {
+    receiverDomainId: number;
+    receiverAddress: string;
+    gas: string;
+  };
+};
+export type ResEnrollRemoteRouter = {
+  receiverDomainId: number;
+};
+
+export type ReqUnenrollRemoteRouter = {
+  signer: string;
+  tokenId: string;
+  receiverDomainId: number;
+};
+export type ResUnenrollRemoteRouter = {
+  receiverDomainId: number;
+};
+
+export type ReqRemoteTransfer = {
+  signer: string;
+  tokenId: string;
+  destinationDomainId: number;
+  recipient: string;
+  amount: string;
+  customHookId: string;
+  gasLimit: string;
+  customHookMetadata: string;
+  maxFee: { denom: string; amount: string };
+};
+export type ResRemoteTransfer = {
+  messageId: string;
+};
+
+export type ResSignAndBroadcast = {
+  height: number;
+  transactionHash: string;
+};
+
+export interface IProvider<T = any> {
```

Comment:
Would it make sense to split this interface into smaller, more focused interfaces. For example, we could have core-related and warp-related functions grouped together into smaller interfaces. Or we can go further and have something like Mailbox, ISM, Hook, Token, Rotuer interfaces.

Reply:
I think the difference in the startup time can be hardly measured with having 100 (uncompiled) lines of code more or less. If we really want to make things faster we should consider only loading code for the relevant chains (e.g. only don't load the 10,000+ lines of code and deps for evm if we only warp deploy cosmos chains).

Regardless of course splitting this up is generally considered better code practice, but for the sake of merging this PR this week I would love to move the discussion of how to split the interface into which groups to the multi-vm design doc if that is fine for you